cmd/azure.go

```diff
@@ -151,7 +151,7 @@ func (a AzureObjects) StorageInfo() StorageInfo {
 }
 
 // MakeBucket - Create a new container on azure backend.
-func (a AzureObjects) MakeBucket(bucket string) error {
+func (a AzureObjects) MakeBucket(location, bucket string) error {
```

Comment:
That will imply a lot of changes:

* ObjectLayer cannot be implemented anymore within GatewayLayer, because MakeBucket has different signature within ObjectLayer and GatewayLayer and cannot override. So need to copy all methods from interface ObjectLayer to GatewayLayer
* globalObjectAPI cannot be used anymore, because we have ObjectLayer and GatewayLayer difererent signature for MakeBucket, need to find other solution for that.

Now I've chosen to add a MakeBucketWithLocation to comply to GatewayLayer interface and MakeBucket to comply to ObjectLayer interface which will call the default region. The MakeBucket will be never called because PutBucket has been overridden for the GatewayLayer.